### PR TITLE
Issue #SB-5905 feat: Reviewer should be able to see the ownership and credits

### PR DESCRIPTION
**Pull request by raghuv45**

### CODE REVIEW

File: src/app/client/src/app/modules/shared/components/content-credits/content-credits.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(ContentCreditsComponent);
+    component = fixture.componentInstance;
+  });
+  it('should take content credits as  INPUT  ', () => {
+    component.contentCreditsData = Response.contentCreditsData;
+    expect(component.showContentCreditModal).toBeDefined();
+    expect(component.showContentCreditModal).toBeFalsy();
+    expect(component.contentCreditsData).toBeDefined();
```
Comment (harishgilimi) [inline src/app/client/src/app/modules/shared/components/content-credits/content-credits.component.spec.ts]: Test cases should contain proper assertions

File: src/app/client/src/app/modules/core/components/content-player-metadata/content-player-metadata.component.html [hunk 2]
```diff
@@ -239,5 +239,22 @@
...
+            <div class="ui grid">
+                <div class="three wide column">
+                    <div class="content content-detail">
+                        {{resourceService.frmelmnts.lbl.ownership}}
+                    </div>
+                </div>
+                <div class="nine wide column">
+                    <!-- *ngIf="metadata.owner" -->
+                    <div class="content" >
+                        sdfsdfsdfdsf {{metadata.owner}} <a (click)="showContentCredits()">Content Credits</a>
```
Comment (harishgilimi) [inline src/app/client/src/app/modules/core/components/content-player-metadata/content-player-metadata.component.html]: why  unnecessary data is in before data binding and what happens when credits or owner are not available

File: src/app/client/src/app/modules/core/components/content-player-metadata/content-player-metadata.component.ts [hunk 3]
```diff
@@ -17,6 +17,8 @@ export class ContentPlayerMetadataComponent implements OnInit, OnDestroy {
   conceptNames: any;
   filteredConcepts: any;
   conceptDataSubscription: Subscription;
+  contentCreditsData: any;
```
Comment (harishgilimi) [inline src/app/client/src/app/modules/core/components/content-player-metadata/content-player-metadata.component.ts]: Why any type for **contentCreditsData** property it should be Object or Array of Objects

File: src/app/client/src/app/modules/core/components/content-player-metadata/content-player-metadata.component.ts [hunk 4]
```diff
@@ -42,6 +44,21 @@ export class ContentPlayerMetadataComponent implements OnInit, OnDestroy {
     });
   }
 
+  showContentCredits () {
+    const contentCredits = _.get(this.metadata, 'content-credits');
+    this.contentCreditsData = { contributors: '', creators: ''};
+    if (contentCredits && contentCredits.length) {
+      this.contentCreditsData.contributors = _.map(contentCredits, 'name').toString();
```
Comment (harishgilimi) [inline src/app/client/src/app/modules/core/components/content-player-metadata/content-player-metadata.component.ts]: **_.map(contentCredits, 'name').toString();** should be **_.toSting( _.map(contentCredits, 'name'));**

File: src/app/client/src/app/modules/core/components/content-player-metadata/content-player-metadata.component.ts [hunk 5]
```diff
@@ -42,6 +44,21 @@ export class ContentPlayerMetadataComponent implements OnInit, OnDestroy {
...
+    }
+    if (this.metadata && this.metadata.owner) {
+      this.contentCreditsData.contributors += this.metadata.owner;
+    }
+    if (this.metadata && this.metadata.creator) {
+      this.contentCreditsData.creators = this.metadata.creator;
```
Comment (harishgilimi) [inline src/app/client/src/app/modules/core/components/content-player-metadata/content-player-metadata.component.ts]: This should contain collaborators as well if available

File: src/app/client/src/app/modules/shared/components/collection-player-metadata/collection-player-metadata.component.html [hunk 6]
```diff
@@ -65,4 +65,18 @@
       </div>
     </div>
   </div>
+  <div class="three wide column">
+    <div class="ui list">
+      <div class="item">
+        <div class="content">
+          <div class="header">Owner</div>
```
Comment (harishgilimi) [inline src/app/client/src/app/modules/shared/components/collection-player-metadata/collection-player-metadata.component.html]: **Owner**  Label should be coming from resource bundles

File: src/app/client/src/app/modules/shared/components/collection-player-metadata/collection-player-metadata.component.html [hunk 7]
```diff
@@ -65,4 +65,18 @@
...
+          <div ng-if="metaData.owner" class="description padding-top-1rem">{{metaData.owner}}
+            <a (click)="showContentCredits()">Content Credits</a>
```
Comment (harishgilimi) [inline src/app/client/src/app/modules/shared/components/collection-player-metadata/collection-player-metadata.component.html]: Cross check all html templates all the text should be coming from resource bundles

File: src/app/client/src/app/modules/shared/components/collection-player-metadata/collection-player-metadata.component.spec.ts [hunk 8]
```diff
@@ -35,4 +37,13 @@ describe('CollectionPlayerMetadataComponent', () => {
...
     expect(component.collectionId).toBe('LP_FT_TextBook2');
   });
+
+  it('should open popup on click of show content credits with contentCredits data', () => {
+    component.metaData = Response.metaData;
+    spyOn(component, 'showContentCredits').and.callThrough();
+    component.showContentCredits();
+    fixture.detectChanges();
+    expect(component.contentCreditsData).toBeDefined();
+    expect(component.showContentCreditsModal).toBeTruthy();
```
Comment (harishgilimi) [inline src/app/client/src/app/modules/shared/components/collection-player-metadata/collection-player-metadata.component.spec.ts]: More test assertions are required as per the implementation and test cases should cover all the possibilities of the code

File: src/app/client/src/app/modules/shared/components/collection-player-metadata/collection-player-metadata.component.ts [hunk 9]
```diff
@@ -1,21 +1,43 @@
...
+
   constructor(public resourceService: ResourceService, private activatedRoute: ActivatedRoute) { }
 
   ngOnInit() {
     this.activatedRoute.params.subscribe((params) => {
       this.collectionId = params.collectionId;
     });
   }
+
+  showContentCredits () {
```
Comment (harishgilimi) [inline src/app/client/src/app/modules/shared/components/collection-player-metadata/collection-player-metadata.component.ts]: Check review comments and if same code written in two components it should be moved to service to not get into **DRY** code anti pattern

**Comment by harishgilimi** [comment]
@raghuv45  reviewed please fix the basic issues and resolve the comments

File: src/app/client/src/app/modules/core/components/content-player-metadata/content-player-metadata.component.html (hunk 2 above)
Comment (raghuv45) [reply src/app/client/src/app/modules/core/components/content-player-metadata/content-player-metadata.component.html]: this change while testing i kept it and missed u remove it.
content credits ui will be shown only when owner is available
when there are no content credits in popup only creator will be shown

File: src/app/client/src/app/modules/shared/components/content-credits/content-credits.component.spec.ts (hunk 1 above)
Comment (raghuv45) [reply src/app/client/src/app/modules/shared/components/content-credits/content-credits.component.spec.ts]: done

File: src/app/client/src/app/modules/shared/components/collection-player-metadata/collection-player-metadata.component.spec.ts (hunk 8 above)
Comment (raghuv45) [reply src/app/client/src/app/modules/shared/components/collection-player-metadata/collection-player-metadata.component.spec.ts]: done

**Pull request merged**